src/Workspaces/CoreTestUtilities/TextReaderTestBase.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public abstract class TextReaderTestBase
+    {
+        [Fact]
+        public void PeakRead()
+        {
+            using var _ = CreateReaders("text", out var referenceReader, out var reader);
+            AssertAllEqual('t', referenceReader.Read(), reader.Read());
+            AssertAllEqual('e', referenceReader.Peek(), reader.Peek());
+            AssertAllEqual('e', referenceReader.Read(), reader.Read());
+            AssertAllEqual('x', referenceReader.Read(), reader.Read());
+            AssertAllEqual('t', referenceReader.Read(), reader.Read());
+            AssertAllEqual(-1, referenceReader.Peek(), reader.Peek());
```

Comment:
I get that testing against a reference `TextReader` implementation is useful, but basically I don't like the idea of removing the expected values, because then you can't see if we're testing reading at the end for example to get -1. If these lines only contained checking against the reference implementation, you wouldn't see whether we're testing interesting cases like that.